lib/asciidoctor-bibtex/js/citeproc.rb

```diff
@@ -0,0 +1,58 @@
+module AsciidoctorBibtex
+  `const {Cite, plugins} = require('@citation-js/core')`
+  `require('@citation-js/plugin-csl')`
+  `const Fs = require('fs')`
```

Comment:
it's better to use Ruby since we provide an implementation depending on the environment

Reply:
You mean instead of `fs`? The problem is that opal doesn't implement `FileTest`. That's why I wrote this directly in javascript.

Reply:
Correct.

> The problem is that opal doesn't implement FileTest. That's why I wrote this directly in javascript.

Yes, but `File` is implemented, you can use `::File.file? 'path'`: https://ruby-doc.org/core-2.5.0/File.html#method-c-file-3F

Reply:
What is the difference compared to FileTest. Can I replace the usage in processor.rb with it as well?

Reply:
I've never used `FileTest` but I guess it's a subset of `File`...?

> Can I replace the usage in processor.rb with it as well?

I would say yes.